src/services/item/plugins/geolocation/service.ts

```diff
@@ -78,14 +78,17 @@ export class ItemGeolocationService {
     // TODO optimize?
     return geoloc
       .map((g) => {
-        if (g.item.id in itemMemberships.data) {
+        const itemId = g.item.id;
+        if (
+          // accessible items
+          itemId in itemMemberships.data &&
+          // accept public items within parent item
+          // otherwise the actor should have at least read permission on root
+          (query.parentItemId || itemMemberships.data[g.item.id])
```

Comment:
Is `itemMemberships.data[g.item.id]` different to `itemId in itemMemberships.data` ? I don't think so, you may have to merge the two conditions ?

If not, you could reuse the `itemId`:

```suggestion
          (query.parentItemId || itemMemberships.data[itemId])
```

Reply:
The difference is that `itemId in itemMemberships.data` is `true` even if `itemMemberships.data[itemId]` is `null`.  This happens if the item is public. I added a comment because it's not that clear indeed!